packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -576,22 +597,60 @@ class ImageCacheStatus {
   String toString() => '${objectRuntimeType(this, 'ImageCacheStatus')}(pending: $pending, live: $live, keepAlive: $keepAlive)';
 }
 
-class _CachedImage {
-  _CachedImage(this.completer, this.sizeBytes);
+/// Base class for [_CachedImage] and [_LiveImage].
+///
+/// Exists primarily so that a [_LiveImage] cannot be added to the
+/// [ImageCache._cache].
+abstract class _CachedImageBase {
+  _CachedImageBase(
+    this.completer, {
+    this.sizeBytes,
+  }) : assert(completer != null),
+       handle = completer.keepAlive();
 
   final ImageStreamCompleter completer;
   int? sizeBytes;
+  ImageStreamCompleterHandle? handle;
+
+  @mustCallSuper
+  void dispose() {
+    assert(handle != null);
+    // Give any interested parties a chance to listen to the stream before we
+    // potentially dispose it.
+    // scheduleMicrotask(() {
+    SchedulerBinding.instance!.addPostFrameCallback((Duration timeStamp) {
```

Comment:
addPostFrameCallback doesn't request a new frame. Are you guaranteed that this code is called only during a frame?

Reply:
No, but we want to keep it alive until the next scheduled frame to give other clients time to subscribe. 

The downside is that if you precache an image and then never ever schedule a frame again, you'll hold the memory. But I'm not clear on how that would happen - you shouldn't be precaching images if you're not going to schedule frames to draw them.

Reply:
k